src/content/reference/react/cache.md

```diff
@@ -0,0 +1,485 @@
+    </>
+  );
+}
+```
+In this example, `App` renders two `WeatherReport`s with the same record. Even though both components do the same work, they cannot share work. `useMemo` is only a local cache to the component instance. 
+
+However, `useMemo` does ensure that if `App` re-renders and the `record` object doesn't change, each component instance would skip work and use the memoized value of `avgTemp`. `useMemo` will only cache the last computation of `avgTemp` with the given dependencies. 
+
+#### `cache` {/*deep-dive-cache*/}
+
+In general, you should use `cache` in Server Components to memoize work that can be shared across components.
+
+```js [[1, 12, "<WeatherReport city={city} />"], [3, 13, "<WeatherReport city={city} />"], [2, 1, "cache(fetchReport)"]]
+const cachedFetchReport = cache(fetchReport);
+
+function WeatherReport({city}) {
+  const report = cachedFetchReport(city);
+  // ...
+}
+
+function App() {
+  const city = "Los Angeles";
+  return (
+    <>
+      <WeatherReport city={city} />
+      <WeatherReport city={city} />
+    </>
+  );
+}
+```
+Re-writing the previous example to use `cache`, in this case the <CodeStep step={3}>second instance of `WeatherReport`</CodeStep> will be able to skip duplicate work and read from the same cache as the <CodeStep step={1}>first `WeatherReport`</CodeStep>. Another difference from the previous example is that `cache` is also recommended for <CodeStep step={2}>memoizing data fetches</CodeStep>, unlike `useMemo` which should only be used for computations.
+
+At this time, `cache` should only be used in Server Components and the cache will be invalidated across server requests.
+
+#### `memo` {/*deep-dive-memo*/}
+
+You should use [`memo`](reference/react/memo) to prevent a component re-rendering if its props are unchanged.
+
+```js
+'use client';
+
+function WeatherReport({record}) {
+  const avgTemp = calculateAvg(record); 
+  // ...
+}
+
+const MemoWeatherReport = memo(WeatherReport);
+
+function App() {
+  const record = getRecord();
+  return (
+    <>
+      <MemoWeatherReport record={record} />
+      <MemoWeatherReport record={record} />
+    </>
+  );
+}
+```
+
+In this example, both `MemoWeatherReport` components will call `calculateAvg` when first rendered. However, if `App` re-renders, with no changes to `record`, none of the props have changed and `MemoWeatherReport` will not re-render. 
+
+Compared to `useMemo`, `memo` memoizes the component render based on props vs. specific computations. Similar to `useMemo`, the memoized component only caches the last render with the last prop values. Once the props change, the cache invalidates and the component re-renders.
+
+</DeepDive>
+
+---
+
+## Troubleshooting {/*troubleshooting*/}
+
+### My memoized function still runs even though I've called it with the same arguments {/*memoized-function-still-runs*/}
+
+Beyond the mentioned Pitfalls, it may be a problem with how React checks if something exists in cache.
```

Comment:
Could you link to the relevant Pitfalls with descriptions of when to follow their advice or remove this sentence? It feels a bit strange to mention there might be relevant answers to someone who finds this section but not to at least help them find the relevant info.